chore: Add MkDocs documentation setup

magicalyak opened the pull request:
## Summary

This PR adds MkDocs support to organize our existing documentation into a searchable website format.

## Changes
- Organized existing docs into `docs/` directory structure
- Added MkDocs configuration with Material theme
- Added development dependencies in `pyproject.toml`
- All content is existing documentation, just reorganized

## Testing
```bash
# Install dependencies
pip install -e ".[dev]"

# Build and serve docs
mkdocs serve
```

## Benefits
- Searchable documentation
- Clean navigation structure  
- Professional appearance
- Easy to deploy as static site

The documentation uses only existing content - no AI-generated material.

magicalyak committed:
Add MkDocs documentation setup

- Organize existing docs into docs/ directory
- Add mkdocs.yml with Material theme
- Add pyproject.toml for dev dependencies
- Keep all original content, just reorganized for better navigation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/netboxlabs/diode?pullRequest=382) <br/>All committers have signed the CLA.

mfiedorowicz commented on pyproject.toml:
```diff
@@ -0,0 +1,34 @@
+version = "1.2.0"
+description = "Diode is a data ingestion service for NetBox that greatly simplifies and enhances the process of adding and updating data in NetBox"
+readme = "README.md"
+requires-python = ">=3.8"
```
nit: python 3.8 reached EOL in October 2024, can we bump it to 3.10 as minimum, please?

magicalyak replied:
Done

mfiedorowicz commented on docs/protocol/diode-proto.md:
Can we rename `protocol` directory to `protobuf`, please?

magicalyak replied:
done

mfiedorowicz commented on mkdocs.yml:
```diff
@@ -0,0 +1,42 @@
+nav:
+    - Introduction: 'index.md'
+    - Getting Started: 'getting-started.md'
+    - Protocol Reference: 'protocol/diode-proto.md'
```
```suggestion
    - Protobuf Reference: 'protobuf/diode-proto.md'
```

magicalyak replied:
Done

mfiedorowicz commented on docs/monitoring/metrics.md:
Can we rename `monitoring` directory to `observability`, please?

magicalyak replied:
Done

mfiedorowicz commented on mkdocs.yml:
```diff
@@ -0,0 +1,42 @@
+    - Introduction: 'index.md'
+    - Getting Started: 'getting-started.md'
+    - Protocol Reference: 'protocol/diode-proto.md'
+    - Monitoring: 'monitoring/metrics.md'
```
```suggestion
    - Observability: 'observability/metrics.md'
```

magicalyak replied:
Done

mfiedorowicz reviewed (commented):
Thanks @magicalyak, a few asks/suggestions. PR also requires CLA to be signed, see https://github.com/netboxlabs/diode/pull/382#issuecomment-3058515679

magicalyak committed:
Address PR feedback

- Update minimum Python version to 3.10 (3.8 reached EOL)
- Rename 'Protocol Reference' to 'Protobuf Reference'
- Rename monitoring directory to observability

magicalyak committed:
Rename protocol directory to protobuf

- Move docs/protocol/ to docs/protobuf/ for consistency
- Update navigation path to match directory name

magicalyak committed:
Fix markdown formatting in tests README

- Wrap <output_folder>/report.html in backticks to prevent HTML interpretation

magicalyak committed:
Fix markdown formatting for placeholder values

- Wrap angle bracket placeholders in backticks to prevent HTML interpretation
- Apply fixes to both docs/getting-started.md and GET_STARTED.md
- Ensures proper rendering in markdown parsers

magicalyak committed:
Revert "Fix markdown formatting for placeholder values"

This reverts commit 7ac1bef5c2071941e8c164f1b25361523ee028a1.

magicalyak commented:
I also fixed one processing issue with a README.md in the tests folder, the other ones aren't needed but that one was breaking processing a bit so this should be good at this point

mfiedorowicz reviewed (approved):
👍

mfiedorowicz commented:
> I also fixed one processing issue with a README.md in the tests folder, the other ones aren't needed but that one was breaking processing a bit so this should be good at this point

FYI existing `/tests` got deprecated with its old integration tests, needs rework with more comprehensive tests soon.
